lib/Transforms/InstCombine/InstCombineVectorOps.cpp

```diff
@@ -1249,7 +1249,17 @@ Instruction *InstCombiner::visitShuffleVectorInst(ShuffleVectorInst &SVI) {
 
   // If the result mask is equal to one of the original shuffle masks,
   // or is a splat, do the replacement.
-  if (isSplat || newMask == LHSMask || newMask == RHSMask || newMask == Mask) {
+  //
+  // XXX EMSCRIPTEN: Add '|| true' so that we always do the replacement.
+  // We're targetting SIMD.js, so there's less of an expectation that a
+  // particular shuffle mask will always map onto a particular instruction on
+  // a particular ISA because we aren't targetting a particular ISA (what the
+  // JS engine does is another story). We may wish to re-evaluate this choice
+  // as we move on to higher-element-count vectors, but especially for now this
+  // is quite desirable.
+  if (isSplat || newMask == LHSMask || newMask == RHSMask || newMask == Mask ||
+      true)
+  {
```

Comment:
This seems pretty suspicious in the sense that it's a change made with the assumption that LLVM is *only* generating asmjs. I wonder if we should back this out?